dev/com.ibm.ws.webserver.plugin.runtime/src/com/ibm/ws/webserver/plugin/runtime/listeners/GeneratePluginConfigListener.java

```diff
@@ -274,7 +277,9 @@ private void submitGeneratePluginTask() {
             Tr.debug(this, tc, "submitGeneratePluginTask : FrameworkState.isStopping() = " + FrameworkState.isStopping());
 
         ExecutorService currentExecutor = executorSrvc;
-        if (!FrameworkState.isStopping() && gpc != null && currentExecutor != null) {
+        final CheckpointPhase checkpoint = CheckpointPhase.getPhase();
```

Comment:
While this is technically a smaller execution path, I find it confusing to put all this logic into a single if block here.

The code would be more clear to me if the original `if` statement was unchanged and instead you short-circuit below in the `if (checkpoint != CheckpointPhase.INACTIVE)` block below like this:

```java
            if (checkpoint != CheckpointPhase.INACTIVE) {
                if (Boolean.valueOf(bc == null ? null : bc.getProperty(CHECKPOINT_GEN_PLUGIN))) {
                    CheckpointHook restoreHook = new CheckpointHook() {
                        @Override
                        public void restore() {
                            currentExecutor.submit(generatePluginTask);
                        }
                    };
                    if (!checkpoint.addMultiThreadedHook(restoreHook)) {
                        currentExecutor.submit(generatePluginTask);
                    }
                }
            }
```